Mathlib/Topology/Separation.lean

```diff
@@ -129,6 +147,55 @@ theorem separatedNhds_iff_disjoint {s t : Set X} : SeparatedNhds s t ↔ Disjoin
 
 alias ⟨SeparatedNhds.disjoint_nhdsSet, _⟩ := separatedNhds_iff_disjoint
 
+/-- `HasSeparatingCover`s can be useful witnesses for `SeparatedNhds`. -/
+def HasSeparatingCover : Set X → Set X → Prop := fun s t ↦
+  ∃ u : ℕ → Set X, s ⊆ ⋃ n, u n ∧ ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) t
+
+/-- Used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+theorem has_separating_covers_iff_separated_nhds {h k : Set X} :
+    HasSeparatingCover h k ∧ HasSeparatingCover k h ↔ SeparatedNhds h k := by
+  constructor
+  · rintro ⟨⟨u, u_cov, u_props⟩, ⟨v, v_cov, v_props⟩⟩
+    have open_lemma : ∀ (u₀ a : ℕ → Set X), (∀ n, IsOpen (u₀ n)) →
+      IsOpen (⋃ n, u₀ n \ closure (a n)) := fun _ _ u₀i_open ↦
+        isOpen_iUnion fun i ↦ (u₀i_open i).sdiff isClosed_closure
+    have cover_lemma : ∀ (h₀ : Set X) (u₀ v₀ : ℕ → Set X),
+        (h₀ ⊆ ⋃ n, u₀ n) → (∀ n, Disjoint (closure (v₀ n)) h₀) →
+        (h₀ ⊆ ⋃ n, u₀ n \ closure (⋃ m ≤ n, v₀ m)) :=
+        fun h₀ u₀ v₀ h₀_cov dis x xinh ↦ by
+      rcases h₀_cov xinh with ⟨un , ⟨n, rfl⟩ , xinun⟩
+      simp only [mem_iUnion]
+      refine ⟨n, xinun, ?_⟩
+      simp_all only [closure_iUnion₂_of_finite, disjoint_right, mem_setOf_eq, mem_iUnion,
+        exists_false, exists_const, not_false_eq_true]
+    refine ⟨
```

Comment:
This is still not following https://leanprover-community.github.io/contribute/style.html. The opening ⟨ should not be orphaned. Please see the last paragraph before https://leanprover-community.github.io/contribute/style.html#instances

Reply:
I think I've corrected this to follow the style guide. (Aside, it's unforunate that this style causes the first argument to be offset horizontally from its sibling arguments.)

3a2386b

Reply:
No it doesn’t offset. Did you look precisely where I said?

Reply:
I believe so? 

![Screenshot_20240615-160724](https://github.com/leanprover-community/mathlib4/assets/1559632/e926434d-ba6c-47c0-917c-feca3cc6528a)

Reply:
This might make my concern clearer.

![Screenshot_20240615-161058](https://github.com/leanprover-community/mathlib4/assets/1559632/320554a6-18b2-4c75-a76f-03099afeaa78)

Reply:
On mobile but I think I realized what's happening: my editor and the docs webpage (on both my laptop and phone browser) are probably rendering that angle bracket as half-width. I can make it align, pretending it was rendered full width.